### PR TITLE
feat: play c8 updates

### PR DESCRIPTION
**Pull request by huygur**
## Description

This PR adds some necessary information regarding Play C8 migration, and updates existing documentation to fit the current behavior.

closes https://github.com/camunda/play/issues/529

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [x] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [x] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Comment by huygur** [comment]
@HanselIdes one more open question I have about this PR;

We have always updated previous versions too, in addition to the Next folder. This time, since Play will only be available in Self Managed for 8.6, should we only update Next folder? 🤔

**Review by HanselIdes — CHANGES_REQUESTED** [review]
I checked off the requirements that jumped out to me from the list in the "Documentation Planning" section, but I didn't see them all and ran out of time for my review. Can you please doublecheck the [Documentation Planning section](https://github.com/camunda/product-hub/issues/1029#:~:text=Documentation%20Planning) and ensure it's all there (or explain why it shouldn't be there)?

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (yT0n1) [inline docs/components/modeler/web-modeler/play-your-process.md:155]: I think these Names are inconsistent with what we have and also not as easy as they could be (obviously, we'd also need to change that in the deployment modal). 
1. Cluster Endpoint should be Zeebe Gateway Address or simply Zebee Address as this is the endpoint for Zeebe and not the entire cluster.
2. OAuth audience, I think should be Zeebe token audience? This makes it less confusing for which service this is
image
3. Besides I think this should be grouped accordingly (both here and in the modal. All Zeebe related things should come first, then the other stuff. 
4. There also seems to be some issues with capitalization. Eg., Zeebe REST URL rather than  Zeebe rest url ?
WDYT?

**Review by yT0n1 — COMMENTED** [review]
A few observations.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:136]: Even Web Modeler is enabled, you would still need to pass PLAY_ENABLED flag to the webapp image, which wash the issue about. I don't think we would have to enable Web Modeler by default as well, right? But its important Play is there by default, in case Web Modeler starts

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:155]: though this might not be a quick update, as some of these fields are shared. I think we would have to update deployment modal as well 😅 and a lot of tests will fail....

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (yT0n1) [reply docs/components/modeler/web-modeler/play-your-process.md:142]: @huygur No we should not enable web modeler by default. I would simply add something like "[...] is `true` by default for the Docker and Kubernetes distributions. Thus, if you enable Web Modeler, Play will be enabled automatically"
I think it's always good to add context. reading this by itself one could think that play will be accessible with the default config. which it's not, as web modeler isn't enabled

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (yT0n1) [reply docs/components/modeler/web-modeler/play-your-process.md:161]: @huygur let's discuss in the standup. I don't think changing Play's modal would constitute a new feature. So I would argue it's ok to "fix" this between alpha and final release. But, yes, let's discuss this in the team

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:100]: I put a generic message like this:

`Alternatively, for this or more advanced use cases you can modify the process instance from Operate.`

which looks like this in the end
Screenshot 2024-08-28 at 12 08 10

**Comment by huygur** [comment]
hey @HanselIdes ,

> I checked off the requirements that jumped out to me from the list in the "Documentation Planning" section, but I didn't see them all and ran out of time for my review. Can you please doublecheck the https://github.com/camunda/product-hub/issues/1029#:~:text=Documentation%20Planning and ensure it's all there (or explain why it shouldn't be there)?

here are my thoughts for the unchecked items;

>  Secrets come from the cluster

✅  added `In SaaS, Play uses connector secrets from your selected cluster.` to the first section `Opening Play`

>  You can interact with a Play instance through Play, Operate, Tasklist, or API. This allows you to test your own custom frontends or code
 Actions now can be triggered by external systems

✅  Added the following section and screenshot

Screenshot 2024-08-28 at 12 11 00


>  Mocking is best performed by using example data
 Mocking connectors can best be done by using service task placeholders for now


✅  We already had a lot of information regarding example data. Added the green part in the following screenshot


Screenshot 2024-08-28 at 12 12 24

>  Inbound connectors should no longer be a limitation

❓  removed the following part in the screenshot.
Anton also validated this works: `Support inbound connectors as message events (e.g. GitHub webhook)` today.
Did I understand the requirement correctly?

Screenshot 2024-08-28 at 12 13 31

> Start events with forms should display the form (TBC)

✅ Removed the following statement
Screenshot 2024-08-28 at 12 14 34

>  Decision table rule evaluations can be viewed in Operate (no link available yet)

✅  Updated the decision table rule statement to the following


Screenshot 2024-08-28 at 12 15 22

**Comment by HanselIdes** [comment]
> @HanselIdes one more open question I have about this PR;
> 
> We have always updated previous versions too, in addition to the Next folder. This time, since Play will only be available in Self Managed for 8.6, should we only update Next folder? 🤔

Yes. We should think about the Self-Managed versions rather than the SaaS versions

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:100]:
```suggestion
1. Add: Select the flow node where you'd like to initiate a new token, then choose 'Add' from the modification dropdown.
```

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:101]:
```suggestion
2. Cancel: Select the flow node where you'd like to cancel active tokens, then choose 'Cancel' from the modification dropdown.
```

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:102]:
```suggestion
3. Move: Select the flow node from which you'd like to move active tokens, then choose 'Move' from the modification dropdown. Afterward, you'll be able to select a target flow node to which the tokens will be relocated.
```

**Review by HanselIdes — DISMISSED** [review]
I recommended some light changes for readability, but otherwise this is good from a PM perspective and is ready for handoff to DevEx

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:169]: Just as a note also here;

we decided not to squeeze in a rushed change in the Modal, therefore docs will also stay like this. 
it will be improved some time between 8.6-alpha5 - 8.6 release.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:133]: What is your suggestion to include here? 
we have mentioned in some places about external actions and inbound connectors. is there anything else you would like to add?

Screenshot 2024-08-28 at 12 27 54

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:163]: this also needs to be resolved to merge this PR. @HanselIdes  @RomanKostka 
I will put another suggestion, where I ended up copying the instructions in https://signup.camunda.com/docker-compose/, but modifying it so Play is available. please check and let me know what you think.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (yT0n1) [inline docs/components/modeler/web-modeler/play-your-process.md:159]: I would also add another limitation here: I is not possible to customize Play's connection beyond what is specified in the Deployment Modal. (see: https://camunda.slack.com/archives/C04FUSLGD6X/p1723709755441369 and https://github.com/camunda/play/issues/552#issuecomment-2298224924)

**Review by yT0n1 — COMMENTED** [review]
Still a few things I would add. Let me know what you think

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:180]: For the docs to follow the user journey, the setup should be before configuration

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:159]: There should be a reference to the Helm setup. Where should the configuration be? Id expect more customization here, so examples are less meaningful. A link to existing docs would be ideal so we don't have to maintain it

**Review by HanselIdes — COMMENTED** [review]
I added some comments about docket and helm setups

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:161]: I don't think we have a proper documentation here, as raised in https://camunda.slack.com/archives/C06SWGVGQ84/p1724327056710069

and I am not sure if I can provide one myself, as the release has not yet happened and we don't have Play enabled in helm charts for alpha releases. I don't feel comfortable without testing myself and seeing the setup actually works, before providing something here.

**Review by HanselIdes — DISMISSED** [review]
Please ensure Play is enabled by default for the standard Helm chart for the 8.6.0 release - at that point we can document the rest

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (christinaausley) [reply docs/components/modeler/web-modeler/play-your-process.md:161]: > we don't have Play enabled in helm charts for alpha releases

Despite this, do we link to this documentation anywhere in the Self-Managed docs, or is this noted somewhere?

CC @conceptualshark

Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:163]: The `PLAY_ENABLED` option for should be added to the 8.6 [web modeler config docs](https://docs.camunda.io/docs/next/self-managed/modeler/web-modeler/configuration/). You should be able to link out to that configuration for both Docker and Helm and not have to maintain any config on this page, including the following alpha feature section. 

Instead, I'd recommend adding the `PLAY_ENABLED` information (and link to the config docs) at the top of this guide, as a part of "Opening Play," so SM users can immediately get to the page they need for this to run.

**Review by conceptualshark — CHANGES_REQUESTED** [review]
A few questions and suggestions specific to Self Managed. 

My general recommendation re: how to maintain the necessary SM docs to set this up is to move the config off to the Web Modeler [SM guide](https://docs.camunda.io/docs/next/self-managed/modeler/web-modeler/configuration/), and not specify either docker or helm instructions in this page. 

For the scope of this PR, it might be better to remove the Docker/SM instructions, and create a follow-up issue to reorganize this page after these updates are merged. I'd be happy to take that on!

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:51]: > is this build up to date?
no, it isn't, since Web Modeler release didn't happen yet. you can use this image for web-modeler-webapp:

registry.camunda.cloud/web-modeler/modeler-webapp-self-managed:ci-3fe9661e2de3bb1fdc595d25b55c448422d66d2f (you need to update docker-compose/camunda-8.6/docker-compose.yaml file) see screenshot below:

Screenshot 2024-08-28 at 22 13 58

Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:51]: but it will be available with alpha5 release.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:17]: in the deployment Modal that is visible in Self Managed when Play tab is clicked. it asks for Cluster details in order to proceed. thats what I meant by 'specified cluster'. how can we improve this?



https://github.com/user-attachments/assets/caf9a843-9edd-4164-a236-2ba8abba3ac1

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:21]: no, Play doesn't support connector secrets in self managed. here's our open ticket for this: https://github.com/camunda/play/issues/271

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:51]: Thank you! You can disregard any of my questions about functionality limited to that build, since I wasn't able to test it at the time. 👍

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:17]: 👍 Thank you for the runthrough - this works for me!

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:21]: If by "In SaaS," we mean saas-only, should we specify?
```suggestion
In SaaS, Play uses Connector secrets from your selected cluster. Connector secrets are not currently supported in Self-Managed.
```

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:163]: the problem is, https://signup.camunda.com/docker-compose/ page does not contain information about how to run Play. if Web Modeler is started using the instructions in the page, Operate and Tasklist (that was started in the first step `Zeebe, Operate and Tasklist – to automate` , wouldn't be using OAuth, which prevents Play (which uses Operate and Tasklist APIs) to be properly authenticated, therefore it doesn't work. This was the best solution I could find, in order to guarantee there's a working instructions once the docs are out. 

for helm setup , I couldn't find clean instructions to run helm charts with web modeler. although, I can already see  PLAY_ENABLED flag is mentioned in the page you linked (see https://docs.camunda.io/docs/next/self-managed/modeler/web-modeler/configuration/#feature-flags) . only difference is that, feature flag will be true by default.

To be honest, I am not sure how to fix this part of the documentation regarding these conditions. but please feel free to commit some changes if you do find a solution.

regarding this:

> For the scope of this PR, it might be better to remove the Docker/SM instructions, and create a follow-up issue to reorganize this page after these updates are merged. I'd be happy to take that on!

I wanted to at least mention the default configuration values that can be used for the docker setup (that is unfortunately not up-to-date in the link I shared), so users better understand what values to put there

Comment (huygur) [reply docs/components/modeler/web-modeler/play-your-process.md:163]: @conceptualshark the follow up ticket you mentioned might also be about removing the instructions and using a link instead, when there is one. this way we can keep the instructions here for now. wdyt?

**Comment by huygur** [comment]
fyi @yT0n1 could you take over the PR in case some other changes are required from our side since I will be on FTO?

there is one unresolved conversation: https://github.com/camunda/camunda-docs/pull/4200#discussion_r1735222696

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [inline docs/components/modeler/web-modeler/play-your-process.md:169]: I restructured this content a bit. Unlike enabling alpha features for SaaS, we don't (yet?) have a guide on enabling/accessing this content yet for SM, or I think it would be appropriate to link here. 

I don't think detailed configuration instructions are necessary if this feature is enabled by default in the versions it's available in. There should be no need to change the `PLAY_ENABLED` flag unless you want to turn it off, in which case you can read about SM configuration on the linked page. 

I also don't think we should/need to provide specific alpha download instructions in their own section. Upgrading to the latest release, or choosing to specifically access the alpha feature early using the linked distributions, seems like it would be enough (for this page) to let SM users know how to opt in early.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:158]: This may change depending on the permissions. I think Commenter access is also possible.

You should confirm the org permissions (dev or admin?) and project permissions (commenter, editor, or admin?) that are required to view the Play tab

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:159]: This should be removed with 8.6.0

All users will have access to Play, but should be able to toggle (for a limited period) whether new Play or old Play is shown

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:167]: Self-managed does not have the concept of alpha features in the same way as SaaS, but the alpha5 release is still relevant for SM

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [reply docs/components/modeler/web-modeler/play-your-process.md:169]: @conceptualshark here are the different potential setups
- self managed modeler, sass cluster
- docker self managed, full platform installation
-docker self managed web modeler with C8 run 
- alpha helm charts

The idea was to lean on docs for the 8.6 release to make the setup journey less painful. Can you help ensure the experience is good?

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [inline docs/components/modeler/web-modeler/play-your-process.md:179]: The table below describes*

**Review by HanselIdes — COMMENTED** [review]
@conceptualshark I added some thoughts and caught a typo section

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:169]: I am happy to provide support on this experience for 8.6, but I think that's separate to what is required as an update for alpha5, which is this PR's current target. What is the expectation around docs delivery and experience for the upcoming alpha? The SM experience now seems limited to installing the alpha Helm chart, or accessing the alpha docker compose file, so I'm not sure what could be provided in this PR to improve that experience.

My expectation was that this feature would be out of alpha and added to the standard Helm and Docker configurations in 8.6, and so anyone using the latest docker, Helm, or upgrading existing Helm charts would have access to Play automatically (eliminating the need for any additional setup). For anyone needing to update existing Docker configurations, we already reference platform repo for examples in the [SM docs](https://docs.camunda.io/docs/next/self-managed/setup/deploy/local/docker-compose/#web-modeler).

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:158]: @yT0n1 Are you able to confirm the correct permissions here?

Comment (yT0n1) [reply docs/components/modeler/web-modeler/play-your-process.md:158]: For now, yes, though recently, also Commenters have access to Play. We will also limit access to Admin and Developer Organization roles. [But that is not yet implemented](https://github.com/camunda/play/issues/593). So we could leave it for now and revisit this paragraph later?

**Comment by yT0n1** [comment]
@conceptualshark and @HanselIdes let me know if any changes are still required from the engineering side. As far as I could see, there aren't any right now?

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (yT0n1) [reply docs/components/modeler/web-modeler/play-your-process.md:167]: Correct me, but I think the reference to alpha can be stripped from this? It won't be alpha anymore after we release 8.6 and this is the /next docs page, right?

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (yT0n1) [reply docs/components/modeler/web-modeler/play-your-process.md:169]: > My expectation was that this feature would be out of alpha and added to the standard Helm and Docker configurations in 8.6, 
Yes, this should be the case. Should we leave the reference to alpha5 in the docs until then and create a separate PR to bump this to 8.6 when the time comes?

Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md]: @yT0n1 👍 This is my opinion as well - I updated the PR to make the alpha SM requirements a little more clear. As we get closer to 8.6, we can revisit this page for any updates/deletions/additional options required. 

@HanselIdes That said, happy to update this further to support product. I assumed it would be convenient to have the SM alpha instructions referenced in this page for any testing, and can follow-up to remove alpha references before 8.6. If we'd prefer not to have them at all here, just let me know.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (conceptualshark) [reply docs/components/modeler/web-modeler/play-your-process.md:158]: Leaving this as-is for now, and we can update as additional roles are implemented/removed.

**Comment by yT0n1** [comment]
@HanselIdes and @conceptualshark could you give me a thumbs up if this is ready to merge? In which case I would do so, as well as create a follow up ticket in Play's repo to revisit the SM installation for 8.6 final release.

File: docs/components/modeler/web-modeler/play-your-process.md
Comment (HanselIdes) [reply docs/components/modeler/web-modeler/play-your-process.md]: > What is the expectation around docs delivery and experience for the upcoming alpha?

@conceptualshark I care that I can reference eager customers to working documentation starting with 8.6-alpha5 and that the installation journey is streamlined for the 8.6.0 release. The plan of merging this now and then following up with improved installation instructions for 8.6.0 makes sense to me

It seems like you and @yT0n1 have a good plan, so I'll add my approval and we can wrap this up

**Review by yT0n1 — APPROVED** [review]

**Comment by yT0n1** [comment]
@conceptualshark would you be amazing and merge this once it has your ok? I'm gonna be on FTO from tomorrow on and it would be good to have this merged

**Pull request merged**
